src/gameplay/GameStart.js

```diff
@@ -1,47 +1,63 @@
 import React from 'react';
-import {firestore} from '../config/firebase';
+import { firestore } from '../config/firebase';
 import { Modal, Button } from 'react-bootstrap';
 import CountTesting from "../backend/countTesting.js";
+import { startGame } from '../backend/startup';
+import { Redirect } from 'react-router-dom';
+import WaitForHost from "./WaitForHost.js";
 
-var roomName = 'Preet Testing';
+var root = 'root';
 
 function GameStart(props) {
-  const [isClicked, setClicked] = React.useState(false);
-  const [joinGame, setJoin] = React.useState("Join Game!");
-  const [accepted, setAccept] = React.useState(false);
-//  const [currentState, setCurrentState] = React.useState(" has joined the lobby");
-  const handleClick = (event) => {setClicked(true); setJoin("Accepted!"); setAccept(true)}; 
-
   const [players, setPlayers] = React.useState([]);
+	const [redirect, setRedirect] = React.useState(false);
+
+  const handleClick = (event) => {
+    startGame(props.roomName);
+    setRedirect(true);
+  };
+
   React.useEffect(() => {
-    const subscribe = firestore.collection(roomName).onSnapshot((snapshot)=>{
-  	let newPlayers = [];
-  	snapshot.docs.forEach((doc)=>{
-      let playerName = doc.data().name;
-      newPlayers.push(playerName);
-    });
+    const subscribe = firestore.collection(root).doc(props.roomName).collection("players").onSnapshot((snapshot) => {
```

Comment:
are you getting a warning whenever a redirect happens here in the console? Like the react component is unmounted. if so, change the subscribe to unsubscribe. (i thought i changed it and merged it so idk why it didnt show up)